src/components/CollectionCarousel.tsx

```diff
@@ -0,0 +1,233 @@
+"use client";
+
+import { CollectionDto, PreviewTitleDto } from "@/dtos";
+import { styled } from "styled-components";
+
+import { EmblaOptionsType } from "embla-carousel";
+import useEmblaCarousel from "embla-carousel-react";
+import { WheelGesturesPlugin } from "embla-carousel-wheel-gestures";
+import { useCallback, useEffect, useState } from "react";
+import { Poster } from "./atomic";
+
+const CarouselParent = styled.div`
+	display: flex;
+	flex-direction: column;
+	gap: 1rem;
+	align-items: center;
+`;
+
+const Carousel = styled.div`
+	overflow: hidden;
+	--slide-size: 20%;
+	--slide-spacing: 1rem;
+	--thumb-size: 2rem;
+	--thumb-spacing: 0.3rem;
+`;
+
+const CarouselContainer = styled.div`
+	display: flex;
+`;
+
+const SlideContainer = styled.div<{ $scale: number }>`
+	display: flex;
+	flex-direction: column;
+	flex: 0 0 var(--slide-size);
+	gap: 1rem;
+	margin-inline: var(--slide-spacing);
+	max-width: 100%;
+	min-width: 0;
+	transform: scale(${(props) => props.$scale});
+`;
+
+const StyledPoster = styled(Poster)`
+	width: 100%;
+`;
+
+const SlidePoster = styled(StyledPoster)``;
+
+const ThumbCarousel = styled(Carousel)`
+	width: 25%;
+
+	mask-image: linear-gradient(
+		to right,
+		transparent,
+		black 10%,
+		black 90%,
+		transparent
+	);
+`;
+
+const ThumbContainer = styled.div`
+	display: flex;
+	flex-direction: column;
+	flex: 0 0 var(--thumb-size);
+	gap: 1rem;
+	margin-inline: var(--thumb-spacing);
+	max-width: 100%;
+	min-width: 0;
+`;
+
+const ThumbPoster = styled(StyledPoster)<{ selected: boolean }>`
+	border-radius: 0.5rem;
+	${({ selected }) => selected && "border: 1px solid #888;"}
+`;
```

Comment:
```suggestion
const ThumbPoster = styled(StyledPoster)<{ $selected: boolean }>`
	border-radius: 0.5rem;
	${({ $selected }) => $selected && "border: 1px solid #888;"}
`;
```

I think this prop should be prefixed with `$`? Otherwise it'll end up on the HTML element?